### ISSUE TITLE
make error when building perl-5.8.8 on HP-UX 11.11

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#9067]
Migrated from [rt.perl.org#46451](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=46451) (status was 'resolved')

Searchable as RT46451$

**Comment by p5pRT** [comment]
### From johnhirsch@northwesternmutual.com

Hi\. I am experiencing the following error when trying to build perl on
HP\-UX&#8203;:

=======
mon&#8203;:\(/opt/perl\-install/perl\-5\.8\.8\) 182> make
&nbsp;        \`sh  cflags "optimize=''" gv\.o\`  gv\.c
&nbsp;          CCCMD =  gcc \-DPERL\_CORE \-c \-fno\-strict\-aliasing \-pipe
\-Wdeclaration\-after\-statement \-I/usr/local/include \-D\_LARGEFILE\_SOURCE
\-D\_FILE\_OFFSET\_BITS=64   \-Wall
gv\.c&#8203;: In function 'Perl\_amagic\_call'&#8203;:
gv\.c&#8203;:1794&#8203;: error&#8203;: 'copy' undeclared \(first use in this function\)
gv\.c&#8203;:1794&#8203;: error&#8203;: \(Each undeclared identifier is reported only once
gv\.c&#8203;:1794&#8203;: error&#8203;: for each function it appears in\.\)
gv\.c&#8203;:1794&#8203;: error&#8203;: expected '\)' before '\_amg'
gv\.c&#8203;:1794&#8203;: error&#8203;: too few arguments to function 'Perl\_amagic\_call'
\*\*\* Error exit code 1

=======

I have checked newsgroups and have only found a few others who had
experienced the same error\, also on HP\-UX\, but haven't seen any
solutions\. Has anyone been able to determine what's causing this error
in the compile? My configuration is below&#8203;:

======
Summary of my perl5 \(revision 5 version 8 subversion 8\) configuration&#8203;:
&nbsp;  Platform&#8203;:
&nbsp;    osname=hpux\, osvers=11\.11\, archname=PA\-RISC2\.0\-thread\-multi
&nbsp;    uname='hp\-ux mon b\.11\.11 u 9000800 283606548 unlimited\-user license
'
&nbsp;    config\_args='\-Dprefix=/opt/perl\-local \-Dusethreads'
&nbsp;    hint=recommended\, useposix=true\, d\_sigaction=define
&nbsp;    usethreads=define use5005threads=undef useithreads=define
usemultiplicity=define
&nbsp;    useperlio=define d\_sfio=undef uselargefiles=define usesocks=undef
&nbsp;    use64bitint=undef use64bitall=undef uselongdouble=undef
&nbsp;    usemymalloc=n\, bincompat5005=undef
&nbsp;  Compiler&#8203;:
&nbsp;    cc='gcc'\, ccflags ='\-fno\-strict\-aliasing \-pipe
\-Wdeclaration\-after\-statement \-I/usr/local/include \-D\_LARGEFILE\_SOURCE
\-D\_FILE\_OFFSET\_BITS=64'\,
&nbsp;    optimize=' '\,
&nbsp;    cppflags='\-Aa \-D\_\_STDC\_EXT\_\_ \-D\_HPUX\_SOURCE \-fno\-strict\-aliasing
\-Wdeclaration\-after\-statement \-I/usr/local/include'
&nbsp;    ccversion='A\.11\.01\.21'\, gccversion='4\.2\.1'\, gccosandvers=''
&nbsp;    intsize=4\, longsize=4\, ptrsize=4\, doublesize=8\, byteorder=4321
&nbsp;    d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=16
&nbsp;    ivtype='long'\, ivsize=4\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\,
lseeksize=8
&nbsp;    alignbytes=8\, prototype=define
&nbsp;  Linker and Libraries&#8203;:
&nbsp;    ld='/usr/bin/ld'\, ldflags =' \-L/usr/local/lib'
&nbsp;    libpth=/usr/local/lib /lib /usr/lib /usr/ccs/lib
&nbsp;    libs=\-lnsl \-lnm \-lndbm \-lmalloc \-ldld \-lm \-lcrypt \-lsec \-lpthread
\-lc
&nbsp;    perllibs=\-lnsl \-lnm \-lmalloc \-ldld \-lm \-lcrypt \-lsec \-lpthread \-lc
&nbsp;    libc=/lib/libc\.sl\, so=sl\, useshrplib=false\, libperl=libperl\.a
&nbsp;    gnulibc\_version=''
&nbsp;  Dynamic Linking&#8203;:
&nbsp;    dlsrc=dl\_hpux\.xs\, dlext=sl\, d\_dlsymun=undef\, ccdlflags='\-Wl\,\-E
\-Wl\,\-B\,deferred '
&nbsp;    cccdlflags='\+Z'\, lddlflags='\-b \+vnocompatwarnings \-L/usr/local/lib'
======

Any ideas you might have would be greatly appreciated\. Thanks\!\!

This e\-mail and any attachments may contain confidential information of Northwestern Mutual\. If you are not the intended recipient of this message\, be aware that any disclosure\, copying\, distribution or use of this e\-mail and any attachments is prohibited\. If you have received this e\-mail in error\, please notify Northwestern Mutual immediately by returning it to the sender and delete all copies from your system\. Please be advised that communications received via the Northwestern Mutual Secure Message Center are secure\. Communications that are not received via the Northwestern Mutual Secure Message Center may not be secure and could be observed by a third party\. Thank you for your cooperation\.

**Comment by p5pRT** [comment]
### From @tux

On Mon\, 15 Oct 2007 14&#8203;:55&#8203;:01 \-0700\, "johnhirsch@&#8203;northwesternmutual\.com \(via
RT\)" \<perlbug\-followup@&#8203;perl\.org> wrote&#8203;:

> \# New Ticket Created by  johnhirsch@&#8203;northwesternmutual\.com 
> \# Please include the string&#8203;:  \[perl \#46451\]
> \# in the subject line of all future correspondence about this issue\. 
> \# \<URL&#8203;: http&#8203;://rt\.perl\.org/rt3/Ticket/Display\.html?id=46451 >
> 
> 
> Hi\. I am experiencing the following error when trying to build perl on
> HP\-UX&#8203;:
> 
> =======
> mon&#8203;:\(/opt/perl\-install/perl\-5\.8\.8\) 182> make
>         \`sh  cflags "optimize=''" gv\.o\`  gv\.c
>           CCCMD =  gcc \-DPERL\_CORE \-c \-fno\-strict\-aliasing \-pipe
> \-Wdeclaration\-after\-statement \-I/usr/local/include \-D\_LARGEFILE\_SOURCE
> \-D\_FILE\_OFFSET\_BITS=64   \-Wall
> gv\.c&#8203;: In function 'Perl\_amagic\_call'&#8203;:
> gv\.c&#8203;:1794&#8203;: error&#8203;: 'copy' undeclared \(first use in this function\)
> gv\.c&#8203;:1794&#8203;: error&#8203;: \(Each undeclared identifier is reported only once
> gv\.c&#8203;:1794&#8203;: error&#8203;: for each function it appears in\.\)
> gv\.c&#8203;:1794&#8203;: error&#8203;: expected '\)' before '\_amg'
> gv\.c&#8203;:1794&#8203;: error&#8203;: too few arguments to function 'Perl\_amagic\_call'
> \*\*\* Error exit code 1

where does the info below come from? A perl you already installed\, or
from \./myconfig?

I ask\, because on HP\-UX PA\-RISC\, you cannot see on the command line if
the gcc you are using is a 64bit or a 32bit compiler\, and 64bit gcc\-4\.x
on HP\-UX 11\.11 is not reliable for building perl\.

If this is a 64bit compiler\, either fetch a precompiled perl from my
site\, or drop back to gcc\-3\.4\.6\, also available from my site\.
http&#8203;://mirrors\.develooper\.com/hpux/downloads\.html\#hpux1111

If this is a 32bit build\, I don't know off\-hand what is the problem\,
but there might be some conflict between the include files that come
with the gcc distribution you use\, and the development depots that
are installed on your system

\# swlist \-l fileset | grep \-\- \-INC | sort

will show you the \-INC filesets that are installed\. On my 11\.11\, the
most recent versions are

&nbsp;  CDEDevKit\.CDE\-INC           B\.11\.11\.02  Header files
&nbsp;  InternetSrvcs\.INETSVCS\-INC  B\.11\.11     INETSVCS\-INC
&nbsp;  PHCO\_31923\.C\-INC            1\.0         ProgSupport\.C\-INC
&nbsp;  PHKL\_35011\.C\-INC            1\.0         ProgSupport\.C\-INC
&nbsp;  PHNE\_32825\.INETSVCS\-INC     1\.0         InternetSrvcs\.INETSVCS\-INC
&nbsp;  PHNE\_34672\.C\-INC            1\.0         ProgSupport\.C\-INC
&nbsp;  PHSS\_33035\.C\-INC            1\.0         B\.11\.11
&nbsp;  ProgSupport\.C\-INC           B\.11\.11     C\-INC

> =======
> 
> I have checked newsgroups and have only found a few others who had
> experienced the same error\, also on HP\-UX\, but haven't seen any
> solutions\. Has anyone been able to determine what's causing this error
> in the compile? My configuration is below&#8203;:
> 
> ======
> Summary of my perl5 \(revision 5 version 8 subversion 8\) configuration&#8203;:
>   Platform&#8203;:
>     osname=hpux\, osvers=11\.11\, archname=PA\-RISC2\.0\-thread\-multi
>     uname='hp\-ux mon b\.11\.11 u 9000800 283606548 unlimited\-user license
> '
>     config\_args='\-Dprefix=/opt/perl\-local \-Dusethreads'
>     hint=recommended\, useposix=true\, d\_sigaction=define
>     usethreads=define use5005threads=undef useithreads=define
> usemultiplicity=define
>     useperlio=define d\_sfio=undef uselargefiles=define usesocks=undef
>     use64bitint=undef use64bitall=undef uselongdouble=undef
>     usemymalloc=n\, bincompat5005=undef
>   Compiler&#8203;:
>     cc='gcc'\, ccflags ='\-fno\-strict\-aliasing \-pipe
> \-Wdeclaration\-after\-statement \-I/usr/local/include \-D\_LARGEFILE\_SOURCE
> \-D\_FILE\_OFFSET\_BITS=64'\,
>     optimize=' '\,
>     cppflags='\-Aa \-D\_\_STDC\_EXT\_\_ \-D\_HPUX\_SOURCE \-fno\-strict\-aliasing
> \-Wdeclaration\-after\-statement \-I/usr/local/include'
>     ccversion='A\.11\.01\.21'\, gccversion='4\.2\.1'\, gccosandvers=''
>     intsize=4\, longsize=4\, ptrsize=4\, doublesize=8\, byteorder=4321
>     d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=16
>     ivtype='long'\, ivsize=4\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\,
> lseeksize=8
>     alignbytes=8\, prototype=define
>   Linker and Libraries&#8203;:
>     ld='/usr/bin/ld'\, ldflags =' \-L/usr/local/lib'
>     libpth=/usr/local/lib /lib /usr/lib /usr/ccs/lib
>     libs=\-lnsl \-lnm \-lndbm \-lmalloc \-ldld \-lm \-lcrypt \-lsec \-lpthread
> \-lc
>     perllibs=\-lnsl \-lnm \-lmalloc \-ldld \-lm \-lcrypt \-lsec \-lpthread \-lc
>     libc=/lib/libc\.sl\, so=sl\, useshrplib=false\, libperl=libperl\.a
>     gnulibc\_version=''
>   Dynamic Linking&#8203;:
>     dlsrc=dl\_hpux\.xs\, dlext=sl\, d\_dlsymun=undef\, ccdlflags='\-Wl\,\-E
> \-Wl\,\-B\,deferred '
>     cccdlflags='\+Z'\, lddlflags='\-b \+vnocompatwarnings \-L/usr/local/lib'
> ======
> 
> Any ideas you might have would be greatly appreciated\. Thanks\!\!


\-\- 
H\.Merijn Brand         Amsterdam Perl Mongers \(http&#8203;://amsterdam\.pm\.org/\)
using & porting perl 5\.6\.2\, 5\.8\.x\, 5\.10\.x  on HP\-UX 10\.20\, 11\.00\, 11\.11\,
& 11\.23\, SuSE 10\.1 & 10\.2\, AIX 5\.2\, and Cygwin\.       http&#8203;://qa\.perl\.org
http&#8203;://mirrors\.develooper\.com/hpux/            http&#8203;://www\.test\-smoke\.org
&nbsp;                        http&#8203;://www\.goldmark\.org/jeff/stupid\-disclaimers/

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From johnhirsch@northwesternmutual.com

Hi\. Thanks for your response\. It didn't show up in my email\, but I 
found this site so I thought I'd reply here\. Please let me know if 
this is not the proper method in which to respond\.

On Tue Oct 16 03&#8203;:00&#8203;:37 2007\, hmbrand wrote&#8203;:
> where does the info below come from? A perl you already installed\, or
> from \./myconfig?

The info came from \./myconfig
 
> I ask\, because on HP\-UX PA\-RISC\, you cannot see on the command line 
if
> the gcc you are using is a 64bit or a 32bit compiler\, and 64bit gcc\-
4\.x
> on HP\-UX 11\.11 is not reliable for building perl\.

To be honest\, I don't know which one it is\. I installed it directly 
from the depot here&#8203;: http&#8203;://hpux\.cs\.utah\.edu/ but it didn't say 
whether it was 32 or 64 bit\. I had assumed it was 32\. Is there any way 
to tell from the gcc distribution?
 
> If this is a 32bit build\, I don't know off\-hand what is the problem\,
> but there might be some conflict between the include files that come
> with the gcc distribution you use\, and the development depots that
> are installed on your system

I ran the swlist on my system and get the following results\. I'm 
missing some of the filesets that you have\, but the ones I do have 
appear to match the version you have\. Could this problem be due to the 
missing filesets?

mon&#8203;:\(/home/hir0122\) 1> swlist \-l fileset | grep \-\- \-INC | sort
&nbsp;  InternetSrvcs\.INETSVCS\-INC                    B\.11\.11        
INETSVCS\-INC   
&nbsp;  PHCO\_31923\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHKL\_23294\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHKL\_23295\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHKL\_23296\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHKL\_23297\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHKL\_23299\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHKL\_23302\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHKL\_23303\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHKL\_23304\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHKL\_23305\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHKL\_23306\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHKL\_23307\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHKL\_23308\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHKL\_23309\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHKL\_23310\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHKL\_23311\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHKL\_23312\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHKL\_23315\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHKL\_24554\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHKL\_24582\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHKL\_24626\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHKL\_25610\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHKL\_26197\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHKL\_26269\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHKL\_28512\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHKL\_30523\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHKL\_30542\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHKL\_30607\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHKL\_30621\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHKL\_32126\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHKL\_32647\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHKL\_33369\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHKL\_33457\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHKL\_34106\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHKL\_34153\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHKL\_34187\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHKL\_34548\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHKL\_34552\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHKL\_34807\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHNE\_23289\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHNE\_32825\.INETSVCS\-INC                       1\.0            
InternetSrvcs\.INETSVCS\-INC 
&nbsp;  PHNE\_35183\.C\-INC                              1\.0            
ProgSupport\.C\-INC 
&nbsp;  PHSS\_35379\.C\-INC                              1\.0            
B\.11\.11        
&nbsp;  ProgSupport\.C\-INC                             B\.11\.11        C\-
INC          
&nbsp;  gcc\.gcc\-
INC                                                                 
&nbsp;  openssl\.OPENSSL\-INC                           A\.00\.09\.07l    OpenSSL 
Header Files

**Comment by p5pRT** [comment]
### From @nwc10

On Tue\, Oct 16\, 2007 at 09&#8203;:38&#8203;:57AM \+0000\, H\.Merijn Brand wrote&#8203;:
> On Mon\, 15 Oct 2007 14&#8203;:55&#8203;:01 \-0700\, "johnhirsch@&#8203;northwesternmutual\.com \(via
> RT\)" \<perlbug\-followup@&#8203;perl\.org> wrote&#8203;:

> > =======
> > mon&#8203;:\(/opt/perl\-install/perl\-5\.8\.8\) 182> make
> >         \`sh  cflags "optimize=''" gv\.o\`  gv\.c
> >           CCCMD =  gcc \-DPERL\_CORE \-c \-fno\-strict\-aliasing \-pipe
> > \-Wdeclaration\-after\-statement \-I/usr/local/include \-D\_LARGEFILE\_SOURCE
> > \-D\_FILE\_OFFSET\_BITS=64   \-Wall
> > gv\.c&#8203;: In function 'Perl\_amagic\_call'&#8203;:
> > gv\.c&#8203;:1794&#8203;: error&#8203;: 'copy' undeclared \(first use in this function\)
> > gv\.c&#8203;:1794&#8203;: error&#8203;: \(Each undeclared identifier is reported only once
> > gv\.c&#8203;:1794&#8203;: error&#8203;: for each function it appears in\.\)
> > gv\.c&#8203;:1794&#8203;: error&#8203;: expected '\)' before '\_amg'
> > gv\.c&#8203;:1794&#8203;: error&#8203;: too few arguments to function 'Perl\_amagic\_call'
> > \*\*\* Error exit code 1

> If this is a 32bit build\, I don't know off\-hand what is the problem\,
> but there might be some conflict between the include files that come
> with the gcc distribution you use\, and the development depots that
> are installed on your system

I have a vague memory that this symptom showed up before on some platform\, and
it's a preprocessor bug\. Something like the pre\-processor has switched back
into K&R mode and so isn't doing ANSI C macro argument concatenation\.

Is your system's compiler fully up to date?

Nicholas Clark

**Comment by p5pRT** [comment]
### From johnhirsch@northwesternmutual.com

On Sun Oct 21 13&#8203;:47&#8203;:10 2007\, nicholas wrote&#8203;:
> 
> I have a vague memory that this symptom showed up before on some
> platform\, and it's a preprocessor bug\. Something like the pre
> processor has switched back into K&R mode and so isn't doing ANSI 
> C macro argument concatenation\.
> 
> Is your system's compiler fully up to date?
> 
> Nicholas Clark

I would hope that it is\, as I just installed it from the standard 
depot files\. However\, we have another server that has the HP ANSI C 
compiler on it\, so I used that compiler and was able to build perl 
with no problems\.

**Comment by p5pRT** [comment]
### From @tux

The last comment obviously shows that the OP has no interest in chasing 
this as a bug\. The other \(similar\) system "fixed" his bug\.

**Comment by p5pRT** [comment]
@tux - Status changed from 'open' to 'resolved'
